WordPress/src/main/java/org/wordpress/android/ui/posts/PromoDialogAdvanced.java

```diff
@@ -0,0 +1,122 @@
+            return PromoDialogAdvanced.newInstance(this);
+        }
+    }
+
+    protected int mButtonNegativeId;
+    protected int mLinkId;
+
+    protected View.OnClickListener mNegativeButtonOnClickListener;
+    protected View.OnClickListener mLinkOnClickListener;
+
+    protected static PromoDialogAdvanced newInstance(Builder builder) {
+        PromoDialogAdvanced fragment = new PromoDialogAdvanced();
+        Bundle args = new Bundle();
+        args.putInt("drawableId", builder.drawableId);
+        args.putInt("titleId", builder.titleId);
+        args.putInt("descriptionId", builder.descriptionId);
+        args.putInt("linkId", builder.linkId);
+        args.putInt("buttonNegativeId", builder.buttonNegativeId);
+        args.putInt("buttonPositiveId", builder.buttonPositiveId);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public @NonNull Dialog onCreateDialog(Bundle savedInstanceState) {
+        mLinkId = getArguments().getInt("linkId");
+        mButtonNegativeId = getArguments().getInt("buttonNegativeId");
+        return super.onCreateDialog(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.promo_dialog_advanced, container);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        ImageView image = (ImageView) view.findViewById(R.id.promo_dialog_image);
+        image.setImageResource(mDrawableId);
+        LinearLayout imageContainer = (LinearLayout) view.findViewById(R.id.promo_dialog_image_container);
+        imageContainer.setVisibility(DisplayUtils.isLandscape(getActivity()) ? View.GONE : View.VISIBLE);
+
+        WPTextView title = (WPTextView) view.findViewById(R.id.promo_dialog_title);
+        title.setText(mTitleId);
+
+        WPTextView description = (WPTextView) view.findViewById(R.id.promo_dialog_description);
+        description.setText(mDescriptionId);
+
+        WPTextView link = (WPTextView) view.findViewById(R.id.promo_dialog_link);
+        if (mLinkId != 0) {
+            link.setText(mLinkId);
+            link.setOnClickListener(mLinkOnClickListener);
+        } else {
+            link.setVisibility(View.GONE);
+        }
+
+        Button buttonNegative = (Button) view.findViewById(R.id.promo_dialog_button_negative);
+        if (mButtonNegativeId != 0) {
+            buttonNegative.setText(mButtonNegativeId);
+            buttonNegative.setOnClickListener(mNegativeButtonOnClickListener);
+        } else {
+            buttonNegative.setVisibility(View.GONE);
+        }
+
+        Button buttonPositive = (Button) view.findViewById(R.id.promo_dialog_button_positive);
+        buttonPositive.setText(mButtonPositiveId);
+        buttonPositive.setOnClickListener(mPositiveButtonOnClickListener);
```

Comment:
I agree it's not necessary as things currently stand. But, we don't have a way of checking that the layout we're using matches the layout of the parent class. So I left in the setup so it's explicit what view names each dialog expects to be there. This is around where the usefulness of the inheritance pattern wears thin 😞